### PR TITLE
[WIP] New package: qgis-3.x.x 

### PR DESCRIPTION
**Pull request by Nyx70**
**Help wanted**

[https://github.com/void-linux/void-packages/issues/9180](https://github.com/void-linux/void-packages/issues/9180)

building instructions: [https://nýx.ch/prj/qgis/](https://nýx.ch/prj/qgis/)

TODO

- [x] New package: libkml [#26727](https://github.com/void-linux/void-packages/pull/26727)
- [x] New package: libspatialite [#28053](https://github.com/void-linux/void-packages/pull/28053)
- [x] proj: update
- [x] New package: proj-data *
- [x] New package: python3-pyproj
- [x] qscintilla-qt5: update
- [x] New package: python3-PyQt5-Qsci
  - [x] python3-PyQt5-Qsci: x86_64*
  - [ ] python3-PyQt5-Qsci: i686*
  - [ ] python3-PyQt5-Qsci: aarch64*
  - [ ] python3-PyQt5-Qsci: arm*
- [x] New package: python3-OWSLib
- [x] libgdal: update
  - [x] libgdal: postgresql & kml support
  - [x] libgdal: OSM support (spatialite)
  - [ ] libgdal: aarch64
  - [ ] libgdal: arm
- [X] New package: qgis
  - [x] qgis: uptate to 3.16.2
  - [x] qgis: uptate to 3.16.3
  - [x] qgis: temporary SIP5 / SIP6 patch
  - [ ] qgis: uptate to 3.18.0
  - [ ] --- fix FindSIP_v6.patch
  - [ ] qgis-server
- [ ] ...

### CODE REVIEW

**Comment by ar-jan** [comment]
I'd love to see this in Void. What's the status of this pull request? What's the best way to help out? (Still new to Void and XPBS myself).

**Comment by sgn** [comment]
Please tag me for packages built by sip.

**Comment by sgn** [comment]
Please use the correct license for each packages.

File: srcpkgs/libspatialite/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" libtool geos-devel"
+fi
+
+post_install() {
+	vlicense COPYING
+}
+
+libspatialite-devel_package() {
+	depends="$pkgname-${version}_${revision}"
```
Comment (sgn) [inline srcpkgs/libspatialite/template:30]: `s/\$pkgname-/${sourcepkg}>=/`

**Comment by sgn** [comment]
What's the point of `shlibs` file inside each packages?

**Comment by sgn** [comment]
I ended up modify a large part of this PR, I'll take over.

**Comment by Nyx70** [comment]
👁️ [#29398](https://github.com/void-linux/void-packages/pull/29398)

**Pull request closed without merging**